File: Project3_Python/Animals/Sheep.py
from Animal import Animal

from Organism import OrganismsEnum


class Sheep(Animal):
    __DEFAULT_INITIATIVE = 4
    __DEFAULT_STRENGTH = 4

    def __init__(self, world, position, strength=__DEFAULT_STRENGTH):
        super().__init__(world, position, strength)
        self.type_of_organism = OrganismsEnum.Sheep
        self._initiative = self.__DEFAULT_INITIATIVE

    def __str__(self):
        return "Sheep"

File: Project2_Java/src/pl/pg/oop/project2/organisms/animals/Sheep.java
package pl.pg.oop.project2.organisms.animals;

import pl.pg.oop.project2.organisms.Animal;
import pl.pg.oop.project2.organisms.OrganismsEnum;
import pl.pg.oop.project2.organisms.World;

import java.awt.*;

public class Sheep extends Animal {
    private final static int DEFAULT_INITIATIVE = 4;
    private final static int DEFAULT_STRENGTH = 4;
    public Sheep(World world, Point position) {
        super(world, position);
        this.TypeOfOrganism = OrganismsEnum.SHEEP;
        this.initiative = DEFAULT_INITIATIVE;
        this.strength = DEFAULT_STRENGTH;
    }

    public Sheep(World world, Point position, int strength) {
        super(world, position, strength);
        this.TypeOfOrganism = OrganismsEnum.SHEEP;
        this.initiative = DEFAULT_INITIATIVE;
    }

    @Override
    public String toString() {
        return "Sheep";
    }
}
